orchestrator/signer.go

```diff
@@ -4,161 +4,111 @@ import (
-		}),
-	); err != nil {
-		log.WithError(err).Errorln("got error, loop exits")
-		return [32]byte{}, err
+		Injective:         inj,
+		PeggyID:           peggyID,
+		LoopDuration:      defaultLoopDur,
 	}
 
-	log.WithField("id", peggyID.Hex()).Debugln("got peggy ID from Ethereum contract")
+	s.logger.WithField("loop_duration", signer.LoopDuration.String()).Debugln("starting EthSigner...")
 
-	return peggyID, nil
+	return loops.RunLoop(ctx, signer.LoopDuration, func() error {
+		return signer.SignValsetsAndBatches(ctx)
+	})
 }
 
-type ethSignerLoop struct {
+type ethSigner struct {
 	*PeggyOrchestrator
-	loopDuration time.Duration
-	peggyID      common.Hash
-	ethFrom      common.Address
+	Injective    cosmos.Network
+	LoopDuration time.Duration
+	PeggyID      gethcommon.Hash
 }
 
-func (l *ethSignerLoop) Logger() log.Logger {
+func (l *ethSigner) Logger() log.Logger {
 	return l.logger.WithField("loop", "EthSigner")
 }
 
-func (l *ethSignerLoop) Run(ctx context.Context) error {
-	return loops.RunLoop(ctx, l.loopDuration, func() error {
-		if err := l.signNewValsetUpdates(ctx); err != nil {
-			return err
-		}
+func (l *ethSigner) SignValsetsAndBatches(ctx context.Context) error {
+	if err := l.signNewValsetUpdates(ctx); err != nil {
+		return err
+	}
 
-		if err := l.signNewBatch(ctx); err != nil {
-			return err
-		}
+	if err := l.signNewBatch(ctx); err != nil {
+		return err
+	}
 
-		return nil
-	})
+	return nil
 }
 
-func (l *ethSignerLoop) signNewValsetUpdates(ctx context.Context) error {
+func (l *ethSigner) signNewValsetUpdates(ctx context.Context) error {
 	oldestUnsignedValsets, err := l.getUnsignedValsets(ctx)
 	if err != nil {
 		return err
 	}
 
 	if len(oldestUnsignedValsets) == 0 {
-		l.Logger().Debugln("no valset updates to confirm")
+		l.Logger().Infoln("no valset updates to confirm")
 		return nil
 	}
 
 	for _, vs := range oldestUnsignedValsets {
 		if err := l.signValset(ctx, vs); err != nil {
 			return err
 		}
-
-		// todo: in case of multiple updates, we should sleep in between tx (non-continuous nonce)
 	}
 
 	return nil
 }
 
-func (l *ethSignerLoop) signNewBatch(ctx context.Context) error {
+func (l *ethSigner) signNewBatch(ctx context.Context) error {
 	oldestUnsignedTransactionBatch, err := l.getUnsignedBatch(ctx)
 	if err != nil {
 		return err
 	}
 
-	if oldestUnsignedTransactionBatch == nil {
-		l.Logger().Debugln("no batch to confirm")
-		return nil
-	}
-
 	if err := l.signBatch(ctx, oldestUnsignedTransactionBatch); err != nil {
 		return err
 	}
 
 	return nil
 }
 
-func (l *ethSignerLoop) getUnsignedBatch(ctx context.Context) (*types.OutgoingTxBatch, error) {
-	var oldestUnsignedBatch *types.OutgoingTxBatch
-	getOldestUnsignedBatchFn := func() (err error) {
-		// sign the last unsigned batch, TODO check if we already have signed this
-		oldestUnsignedBatch, err = l.inj.OldestUnsignedTransactionBatch(ctx)
-		if errors.Is(err, cosmos.ErrNotFound) || oldestUnsignedBatch == nil {
-			return nil
-		}
```

Comment:
are these cases safe to omit? @dbrajovic

Reply:
The actual [peggy query](https://github.com/InjectiveLabs/injective-core/blob/317c891263837c0ec79fe214f1e83f364275e017/injective-chain/modules/peggy/keeper/grpc_query.go#L144) returns an error only if the address is malformed. So the `ErrNotFound` check never practically triggers. The check was used to ensure no error bubbles up (with `|| oldestUnsignedBatch == nil`) which we still uphold